packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -446,8 +446,8 @@ export default {
           if (def.node != null && def.node.init === node.parent) {
             continue;
           }
-          // Ignore Flow type parameters
-          if (def.type === 'TypeParameter') {
+          // Ignore Flow type parameters and local type aliases
+          if (def.type === 'TypeParameter' || def.node.type === 'TypeAlias') {
```

Comment:
Probably also want to check for `OpaqueType`. The other probably more robust option is i believe the `def` has a property which indicates if its a value def or not.

Reply:
Adding a test for it, but somehow it's not considered as a dependency. Maybe the scope analysis somewhere doesn't pick up the opaque type. Seems pretty unlikely someone needs to define a local opaque type though, so I think a test to catch this in case the scope analysis is updated at some point should be sufficient.